pandas/core/generic.py

```diff
@@ -4418,6 +4422,86 @@ def sort_values(
         """
         raise AbstractMethodError(self)
 
+    def sort_index(
+        self,
+        axis=0,
+        level=None,
+        ascending: bool_t = True,
+        inplace: bool_t = False,
+        kind: str = "quicksort",
+        na_position: str = "last",
+        sort_remaining: bool_t = True,
+        ignore_index: bool_t = False,
+        key: IndexKeyFunc = None,
+    ):
+
+        inplace = validate_bool_kwarg(inplace, "inplace")
+        axis = self._get_axis_number(axis)
+        target = self._get_axis(axis) if isinstance(self, ABCDataFrame) else self.index
```

Comment:
this should be the same for DataFrame/Series (it would raise if passed an incorrect axis for series)